strongbox-commons/src/main/java/org/carlspring/strongbox/booters/SystemPropertiesInitializer.java

```diff
@@ -0,0 +1,57 @@
+package org.carlspring.strongbox.booters;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Ensures that a handful of core system properties are set prior to starting the application.
+ *
+ * @author cbono
+ */
+class SystemPropertiesInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
+{
+    @VisibleForTesting
+    static final String STRONGBOX_HOME_PROPERTY_NAME = "strongbox.home";
```

Comment:
I think we should have an enum with all of the well-known property names. I did not want to bite that off here so I compromised and put them internally here for now.